### PR TITLE
Update training packages to Pytorch 1.11.0

### PR DESCRIPTION
**Pull request by raviskolli**
Update ORT training packages to Pytorch 1.11

### CODE REVIEW

**Review by thiagocrepaldi — DISMISSED** [review]

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+--pre
+-f https://download.pytorch.org/whl/torch_stable.html
+torch==1.11.0
+torchvision==0.12.0
+torchtext==0.12.0
+setuptools>=41.4.0
+diff --git a/tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.9.0_cu10.2.txt b/tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.9.0_cu10.2.txt
```
Comment (ytaous) [inline tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt:7]: >1.9.0 [](http://example.com/codeflow?start=100&length=5)

is this name right? same for the 1.9 on the far right

File: tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+--pre
+-f https://download.pytorch.org/whl/torch_stable.html
+torch==1.11.0
```
Comment (ytaous) [inline tools/ci_build/github/linux/docker/scripts/training/ortmodule/stage1/requirements_torch1.11.0_cu10.2.txt:3]: should this be +cu102? same for torchvision

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3 [hunk 3]
```diff
@@ -182,7 +182,7 @@ CMD ["/bin/bash"]
 
 #Build manylinux2014 docker image end
 ARG PYTHON_VERSION=3.6
-ARG TORCH_VERSION=1.10.0
+ARG TORCH_VERSION=1.11.0
```
Comment (ytaous) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_training_cuda11_3:185]: are we only changing limited set of files, as there are also 11.1 and 11.4, etc.?

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1110-cuda102.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+trigger: none
+
+stages:
```
Comment (ytaous) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1110-cuda102.yml:3]: Do we still keep torch 1.9 yml under the same dir?

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1110-cuda102.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+trigger: none
```
Comment (baijumeswani) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-torch1110-cuda102.yml:1]: Why are we adding this file? How do you feel about leaving the orttrainer tests (old tests being invoked from build.py) depend on torch 1.9 (as they have been so far). We can only change the torch version for ortmodule tests?

File: onnxruntime/python/tools/transformers/gpt2_helper.py [hunk 6]
```diff
@@ -407,7 +407,6 @@ def export_onnx(model,
                           f=onnx_model_path,
                           input_names=input_names,
                           output_names=output_names,
-                          example_outputs=outputs,
```
Comment (snnn) [inline onnxruntime/python/tools/transformers/gpt2_helper.py:410]: My worry is whether the new code will continue to work well with pytorch 1.10 and pytorch 1.9. If the parameter is not necessary, initially we didn't need to have it. Which means there is some garbage code around it that needs be cleaned. Intuitively, I feel it isn't the case.  For what I know, right now we can't drop the support for pytorch 1.10 yet. 

So before this PR gets merged, we need to make it clear which pytorch versions onnxruntime should support.  @askhade  FYI.

Comment (baijumeswani) [reply onnxruntime/python/tools/transformers/gpt2_helper.py:410]: ORT training package has two primary python modules:
1. `ORTTrainer`
2. `ORTModule`

While we continue supporting `ORTModule` with the latest torch version, we probably want to come up with a plan to deprecate `ORTTrainer`. So, we should move torch version to 1.11 for ORTModule, and continue using 1.10 for ORTTrainer until we phase out ORTTrainer and archive it. Until then, we probably will have to support two versions of torch. We can have a more detailed discussion on this.

Comment (harshithapv) [reply onnxruntime/python/tools/transformers/gpt2_helper.py:410]: @snnn yes this code will work with torch 1.10 and torch 1.9.0. We have all the removed example_ouput commits in this PR - https://github.com/microsoft/onnxruntime/pull/10864 and the CIs have passed.

**Review by ytaous — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**